### DISCUSSION TITLE
Missing file update when "make"

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#11192] [ESP32]
Hi there, I'm here writing to you just to understand the reason why when I run "make all" command, without a "make clean" before, I don't get files modifications. In order to get modifications I have to run a make clean before.

When I was using Pycom modules I didn't need to run clean every time because make all was looking for differences and only them were compiled.

Do you have any idea on how to avoid the make clean every time? Thank you very much

**Comment by peterhinch** [comment]
I think there must be something wrong with your build environment. I've never seen this.

**Reply by ghost** [reply]
It may be. Is somebody able to help me finding out what might be wrong? What I forgot to say is that I was talking about modifications on .py files

As an example I just made make clean -> make all -> made some modifications on .py files -> make all -> Modified files were not update, compilation finished soon with no modifications

**Reply by peterhinch** [reply]
You'll need to provide more information for anyone to help here. Where are the Python files located? What manifest files (contents and location) have you defined to ensure that they are included in the build? See [docs](http://docs.micropython.org/en/latest/reference/manifest.html).

**Reply by ghost** [reply]
Hi Peterninch, thanks for your support.
Python files are located into `ports\esp32\modules`

Why do you think it's a wrong manifest file since when I make clean and then make all everything goes up very well? The problem appears when I make some later modifications to the files and I don't make clean but directly make all. Files are not re-compiled with the modifications.

Don't you think that if it's a manifest file I should never get a successful compilation?
Thank you again

**Reply by peterhinch** [reply]
If your source files are on a server and you're building on a PC, problems like this can occur if the server's clock and the PC's clock differ. Make utilities rely on the creation date of source files.

**Reply by ghost** [reply]
Ok @peterhinch , thanks for your tip.
My source files are on my local PC, on a Windows directory shared with a linux virtual machine from where I compile. The windows OS and the virtualized linux OS have the same clock time.
I might try to copy the source files on my linux OS machine e compile them to see if the problem persists.
Do you have any other ideas?
Thank you very much
